src/core/pal/layer.cpp

```diff
@@ -338,6 +338,12 @@ void Layer::joinConnectedFeatures()
     }
   }
   mConnectedHashtable.clear();
+
+  // Expunge feature parts that are smaller than the minimum size required
+  mFeatureParts.erase( std::remove_if( mFeatureParts.begin(), mFeatureParts.end(), []( FeaturePart * part )
+  {
+    return part->feature()->minimumSize() != 0.0 && part->length() < part->feature()->minimumSize();
```

Comment:
Looking at this again, it's actually leaking here because we are only removing the pointers from the list, but not actually deleting them. 

```suggestion
    if ( part->feature()->minimumSize() != 0.0 && part->length() < part->feature()->minimumSize() )
    {
        delete part;
        return true;
    }
    else
    {
      return false;
    }
```

Reply:
(after we merge this i'll go back and see if we can make mFeatureParts a list of unique_ptrs)